storage/src/main/java/org/apache/kafka/storage/internals/log/LogConfig.java

```diff
@@ -424,6 +387,18 @@ public int initFileSize() {
             return 0;
     }
 
+    public boolean remoteStorageEnable() {
+        return remoteLogConfig.remoteStorageEnable;
+    }
+
+    public long localRetentionMs() {
+        return remoteLogConfig.localRetentionMs == LogConfig.DEFAULT_LOCAL_RETENTION_MS ? retentionMs : remoteLogConfig.localRetentionMs;
+    }
+
+    public long localRetentionBytes() {
+        return remoteLogConfig.localRetentionBytes == LogConfig.DEFAULT_LOCAL_RETENTION_BYTES ? retentionSize : remoteLogConfig.localRetentionBytes;
```

Comment:
When will we use them? I think they should be called when retrieved the value since we don't expect to get value of `-2`.

Reply:
yes, correct. The caller should get the real local log retention values when set to -2.